File: 10 - Python/Exercícios/05 - Matrizes/Exercício 01.py
#Exercício - Diagonal Negativos

ordem = int(input("Qual a ordem da matriz? "))

negativos = 0

mat = [[0 for x in range(ordem)] for x in range(ordem)]

for i in range(ordem):
    for j in range(0, ordem):
        mat[i][j] = int(input(f"Elemento [{i},{j}]: "))

        if mat[i][j] < 0:
            negativos = negativos + 1

print("")
print("Diagonal principal:")

for i in range(ordem):
    print(f"{mat[i][i]} ", end="")

print("")
print("Quantidade de negativos =", negativos)

File: 07 - C++/Exercícios/05 - Matrizes/Exercício 01.cpp
//Exercício - Diagonal Negativos

#include <iostream>

using namespace std;

int main(){

    int i, j, ordem, negativos = 0;

    cout << "Qual a ordem da matriz? ";
    cin >> ordem;

    int mat[ordem][ordem];

    for(i = 0; i < ordem; i++){
        for(j = 0; j < ordem; j++){
            cout << "Elemento [" << i  << ", " << j << "]: ";
            cin >> mat[i][j];

            if(mat[i][j] < 0){
                negativos++;
            }
        }
    }

    cout << endl << "Diagonal principal:" << endl;

    for(i = 0; i < ordem; i++){
        cout << mat[i][i] << " ";
    }

    cout << endl << "Quantidade de negativos = " << negativos;

    return 0;

}